File: src/day11.kt
import java.util.Scanner


fun main() {
    val scanner = Scanner(System.`in`)
    val input: List<Long> = scanner.nextLine().split(" ").map{ it.toLong() }
    var cnt: Map<Long, Long> = input.groupingBy { it }.eachCount().mapValues { it.value.toLong() }

    val t = 75

    for (i in 0 until t) {
        val temp = mutableMapOf<Long, Long>()
        for ((k, v) in cnt) {
            if (k == 0L) temp[1] = temp.getOrDefault(1, 0) + v
            else if ( k.toString().length % 2 == 0) {
                val n = k.toString().length / 2
                val left = k.toString().substring(0, n).toLong()
                val right = k.toString().substring(n).toLong()
                temp[left] = temp.getOrDefault(left, 0L) + v
                temp[right] = temp.getOrDefault(right, 0L) + v
            } else {
                temp[k * 2024L] = temp.getOrDefault(k * 2024L, 0L) + v
            }
        }
        cnt = temp
    }
    println(cnt.values.sum())
}

File: python/day11.py
from collections import Counter
import sys

input = sys.stdin.readline

inputs = list(map(int, input().strip().split()))

cnt = Counter(inputs)

T = 75

for _ in range(T):
    temp = Counter()
    for num in cnt:
        if num == 0:
            temp[1] += cnt[num]
        elif (n:=len(str(num)))%2 == 0:
            left, right = str(num)[:n//2], str(num)[n//2:]
            temp[int(left)] += cnt[num]
            temp[int(right)] += cnt[num]
        else:
            temp[num*2024] += cnt[num]
    cnt = temp

print(sum(cnt.values()))